lib/chef/knife/ec2_server_create.rb

```diff
@@ -851,6 +851,44 @@ def eip_scope
         end
       end
 
+      def ssl_config_user_data
+<<-EOH
+
+$vm_name = invoke-restmethod -uri http://169.254.169.254/latest/meta-data/public-ipv4
+winrm quickconfig -q
```

Comment:
We should do this only if winrm is not enabled (probably by checking the status of the winrm service). Also, this creates a plaintext listener -- if we see that there's a new listener in existence after winrm quickconfig -q, we should delete that listener since we don't actually want the plaintext listener.